src/cmap/connection_pool_events.ts

```diff
@@ -194,6 +207,20 @@ export class ConnectionCheckOutFailedEvent extends ConnectionPoolMonitoringEvent
   error?: MongoError;
   /** @internal */
   name = CONNECTION_CHECK_OUT_FAILED;
+  /**
+   * The time it took to check out the connection.
+   * More specifically, the time elapsed between
+   * emitting a `ConnectionCheckOutStartedEvent`
+   * and emitting this event as part of the same checking out.
+   *
+   * Naturally, if a new connection was not created (`ConnectionCreatedEvent`)
+   * and established (`ConnectionReadyEvent`) as part of checking out,
+   * this duration is usually
+   * not greater than `ConnectionPoolOptions.waitQueueTimeoutMS`,
+   * but MAY occasionally be greater than that,
+   * because a driver does not provide hard real-time guarantees.
+   */
```

Comment:
I think we can remove the entire second paragraph here- I think this is mostly relevant for driver implementers and less relevant for driver users.

Reply:
Ah I see, I'll remove it.